File: src/bindings/async_process.cpp
#include "bindings.hpp"

#include <sol/state.hpp>

#include "../async_process.hpp"
// Include required because async_process.hpp forward declares Document.
#include "../document.hpp" // IWYU pragma: keep.

void AsyncProcessBinding::init_bridge(sol::table& core) {
    // clang-format off
    core.new_usertype<AsyncProcess>("AsyncProcess",
        /* Properties. */
        "command", sol::readonly(&AsyncProcess::command_),
        "args", sol::readonly(&AsyncProcess::args_),
        "doc", sol::readonly(&AsyncProcess::doc_),

        /* Functions. */
        "spawn", &AsyncProcess::spawn,
        "kill", &AsyncProcess::kill);
    // clang-format on
}

File: lua/bindings/async_process.lua
--- @meta

--- An asyncronously running process.
--- @class Core.AsyncProcess
--- @field command string The running command.
--- @field args string[] The arguments to the command.
--- @field doc Core.Document The Document the process is writing to.
Core.AsyncProcess = {}

--- Starts a process.
--- @param color boolean Enable or disable (ANSI) color output.
--- @return boolean
function Core.AsyncProcess:spawn(color) end

--- Kills a process.
function Core.AsyncProcess:kill() end
